crypto/verifier/verifier.go

```diff
@@ -1,48 +1,49 @@
-package chain
+package verifier
 
 import (
 	"crypto/sha256"
 
-	"github.com/drand/drand/common/scheme"
-	"github.com/drand/drand/key"
+	"github.com/drand/drand/chain"
+	"github.com/drand/drand/crypto"
 	"github.com/drand/kyber"
 )
 
 // Verifier allows verifying the beacons signature based on a scheme.
 type Verifier struct {
 	// scheme holds a set of values the verifying process will use to act in specific ways, regarding signature verification, etc
-	scheme scheme.Scheme
+	crypto.Scheme
 }
 
-func NewVerifier(sch scheme.Scheme) *Verifier {
-	return &Verifier{scheme: sch}
+func NewVerifier(sch crypto.Scheme) *Verifier {
+	return &Verifier{Scheme: sch}
 }
 
 // DigestMessage returns a slice of bytes as the message to sign or to verify
 // alongside a beacon signature.
 func (v Verifier) DigestMessage(currRound uint64, prevSig []byte) []byte {
```

Comment:
Actually this could also be a scheme method, and probably should. There is very little differentiation now between what a verifier is able to do and what a scheme should do. I would go as far as removing entirely the verifier struct, which only acts a proxy for a scheme now, The scheme can do this job now of verifying digesting etc.
Keeping them separate means there is a tight relationship between the two, i.e. changing one means changing the other all the time. and we generally want to avoid such coupling.

Reply:
We need to have the verifier leave in a different package to avoid cyclic dependencies if we want the verifier to use the `chain` package's stuff. :(